src/constructs/autoscaling/asg.ts

```diff
@@ -21,6 +32,50 @@ export interface GuAutoScalingGroupProps
   overrideId?: boolean;
 }
 
+/**
+ * `minimumCodeInstances` and `minimumProdInstances` determine the number of ec2 instances running
+ * in each stage under normal circumstances (i.e. when there are no deployment or scaling events in progress).
+ *
+ * `maximumCodeInstances` and `maximumProdInstances` are both optional. If omitted, these
+ * will be set to `minimumCodeInstances * 2` and `minimumProdInstances * 2`, respectively.
+ * This allows us to support Riff-Raff's autoscaling deployment type by default.
+ *
+ * The maximum capacity values can be set manually if you need to scale beyond the default limit (e.g. due to heavy traffic)
+ * or restrict scaling for a specific reason.
+ */
+export interface GuAsgCapacityProps {
+  minimumCodeInstances: number;
+  minimumProdInstances: number;
+  maximumCodeInstances?: number;
+  maximumProdInstances?: number;
+}
+
+interface AwsAsgCapacityProps {
+  minCapacity: number;
+  maxCapacity: number;
+}
+
+function wireCapacityProps(stack: GuStack, capacity: GuAsgCapacityProps): AwsAsgCapacityProps {
+  const minInstancesKey = "minInstances";
+  const maxInstancesKey = "maxInstances";
+  const minInstances: GuStageDependentNumber = {
+    variableName: minInstancesKey,
+    codeValue: capacity.minimumCodeInstances,
+    prodValue: capacity.minimumProdInstances,
+  };
+  const maxInstances: GuStageDependentNumber = {
+    variableName: maxInstancesKey,
+    codeValue: capacity.maximumCodeInstances ?? capacity.minimumCodeInstances * 2,
+    prodValue: capacity.maximumProdInstances ?? capacity.minimumProdInstances * 2,
+  };
+  stack.setStageDependentValue(minInstances);
+  stack.setStageDependentValue(maxInstances);
+  return {
+    minCapacity: stack.getStageDependentValue(minInstancesKey),
+    maxCapacity: stack.getStageDependentValue(maxInstancesKey),
+  };
+}
```

Comment:
Will we need to write code like this for every situation where we'll have stage-dependent non-private values to handle? Is there any way to make this easier going forward with helper functions, or should we just accept the responsibility to implement these on a case-by-case basis so that the library users can benefit from the result?

Reply:
I guess the main opportunity for cutting down on boilerplate here would be to provide a single function which sets up the mapping and returns the [`FindInMap`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html) which references it.